### PR TITLE
[RNBW-4464] Fix unit testing setup

### PR DESCRIPTION
**Pull request by estrattonbailey**
Fixes RNBW-4464

## What changed (plus any additional context for devs)
Fixes our Jest setup (non-e2e) and readies it for use. **It does not fix failing tests, it only enables the tests to be run.** 

Also, at the moment, the `react-native` Jest preset is blocked from accessing/mocking native modules, so that is a remaining step we'll need to fix before we can move forward.

This PR contains the rest of the needed changes.

## What to test
It takes an added step to test locally atm. Reach out to Eric in Slack for details.

No runtime code should be affected, and these tests are NOT run in CI.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: config/test/jest-setup.js [hunk 1]
```diff
@@ -21,7 +15,6 @@ jest.mock('@sentry/react-native', () => ({
   captureException: () => null,
 }));
 
-jest.autoMockOff();
```
Comment (estrattonbailey) [inline config/test/jest-setup.js:24]: Auto mock isn't on anymore, so we don't need this

File: jest.config.js [hunk 2]
```diff
@@ -5,9 +5,13 @@ const { compilerOptions } = require('./tsconfig');
 module.exports = {
   preset: 'react-native',
   setupFiles: ['./config/test/jest-setup.js'],
+  testPathIgnorePatterns: ['node_modules', 'e2e'],
```
Comment (estrattonbailey) [inline jest.config.js:8]: Default is `node_modules`, this adds `e2e` to ignored tests

File: jest.config.js [hunk 3]
```diff
@@ -5,9 +5,13 @@ const { compilerOptions } = require('./tsconfig');
...
   transform: {
     '\\.tsx?$': 'ts-jest',
   },
+  transformIgnorePatterns: [
+    'node_modules/(?!((jest-)?react-native|react-native-keyboard-area|@react-native(-community)?)/)',
+  ],
```
Comment (estrattonbailey) [inline jest.config.js:14]: Anything matched here gets transpiled. There are a few react-native libs that ship untranspiled code.

**Review by estebanmino — APPROVED** [review]
unit testing working 💯

**Review by brunobar79 — APPROVED** [review]

**Pull request merged**